lib/redirect.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const { full_url_for } = require('hexo-util');
+
+module.exports = function(data) {
+  const { alias, aliases, redirect } = data;
+  if (redirect && !alias && !aliases) {
+    const target = full_url_for.call(this, redirect);
```

Comment:
can integrate [`post_path`](https://hexo.io/docs/tag-plugins#Include-Posts) tag plugin in future, so that user can specify target post's filename ("foo-bar"), instead of permalink ("2020/01/02/foo-bar").